File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cpp
# include <stack>

class MyQueue {
public:

    MyQueue() {}
    ~MyQueue() {}

    stack <int> myStackFinal;
    stack <int> myStackTemp;

    void push(int x) {
        myStackFinal.push(x);
    }
    
    int pop() {
        int tempTopValue = 0;

        while(myStackFinal.size() > 1){
            tempTopValue = myStackFinal.top();
            myStackTemp.push(tempTopValue);
            myStackFinal.pop();
        }

        tempTopValue = myStackFinal.top();
        myStackFinal.pop();

        while(myStackTemp.size() != 0){
            int newTempTopValue = myStackTemp.top();
            myStackFinal.push(newTempTopValue);
            myStackTemp.pop();
        }
        return tempTopValue;
    }
    
    int peek() {
        while(myStackFinal.size() > 1){
            int tempTopValue = myStackFinal.top();
            myStackTemp.push(tempTopValue);
            myStackFinal.pop();
        }

        int frontValue = myStackFinal.top();

        while(myStackTemp.size() != 0){
            int tempTopValue = myStackTemp.top();
            myStackFinal.push(tempTopValue);
            myStackTemp.pop();
        }
        return frontValue;
    }
    
    bool empty() {
        if(myStackFinal.size() == 0){
            return true;
        }else{
            return false;
        }
    }

};

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.stackOne = []
        
    def push(self, x: int) -> None:
        self.stackOne.append(x)
        
    def pop(self) -> int:
        tempStack = []
        while len(self.stackOne) != 1:
            tempStack.append(self.stackOne.pop())
        answer = self.stackOne[0]
        self.stackOne = []
        while len(tempStack) != 0:
            self.stackOne.append(tempStack.pop())
        return answer

    def peek(self) -> int:
        tempStack = []
        while len(self.stackOne) != 1:
            tempStack.append(self.stackOne.pop())
        answer = self.stackOne[0]
        tempStack.append(self.stackOne[0])
        self.stackOne = []
        while len(tempStack) != 0:
            self.stackOne.append(tempStack.pop())
        return answer

    def empty(self) -> bool:
        if len(self.stackOne) == 0:
            return True
        else:
            return False
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()